R/compile_model.R

```diff
@@ -95,11 +94,14 @@ compile_model <- function(model, name) {
                 
         ## update C_code slot
         model@C_code <- character()
+        
+        ## update comp_name slot
+        model@comp_name <- filename
     } else {
         stop("No C code to compile")
     }
-    ## output new class and model
-    setClass(filename, contains = c("SimInf_model"), where = topenv(parent.frame()))
-    as(model, filename)
+    
+    ## output revised model
+    model
```

Comment:
It's now possible keep the C_code slot unchanged and to return the `SimInf_model_dll` class

```
    ## output revised model
    model <- as(model, "SimInf_model_dll")
    model@filename <- lib
    model@name  <- name
    model
```